webapp/src/Controller/Jury/SubmissionController.php

```diff
@@ -449,7 +450,26 @@ public function viewAction(Request $request, int $submitId)
                 ->getQuery()
                 ->getResult();
 
+            $judgingRunTestcaseIdsInOrder = $this->em->createQueryBuilder()
+                ->from(JudgingRun::class, 'jr')
+                ->select('jr.testcaseid')
+                ->andWhere('jr.judging = :judging')
+                ->setParameter(':judging', $selectedJudging)
+                ->orderBy('jr.endtime')
+                ->getQuery()
+                ->getScalarResult();
+
+            $cnt = 0;
             foreach ($runResults as $runResult) {
+                /** @var Testcase $testcase */
+                $testcase = $runResult[0];
+                if (isset($judgingRunTestcaseIdsInOrder[$cnt])) {
+                    if ($testcase->getTestcaseid() != $judgingRunTestcaseIdsInOrder[$cnt]['testcaseid']) {
```

Comment:
If ``judgingRunTestcaseIdsInOrder[$cnt]` is not set, we should also emit the error right? Maybe remove the `isset` and change the right part of the if to `$judgingRunTestcaseIdsInOrder[$cnt]['testcaseid'] ?? null`?

Reply:
Nope. If we have lazy judging the this list will contain less items than the one we are iterating through and that's fine.